src/components/ec-amount-input/ec-amount-input.vue

```diff
@@ -120,7 +112,13 @@ watch(() => props.modelValue, (newValue) => {
       return;
     }
 
-    formattedValue.value = newValue;
+    if (newValue === null || typeof newValue !== 'string') {
```

Comment:
it means that if we pass number in `props.modelValue` we won't set `formattedValue` to `newValue`? Previous logic was different.

Reply:
Yes, TS flagged that. The problem with this component is that the modelValue is string when isMasked is true and is number | null if isMasked is false. formattedValue should hold the nicely formatted version of the modelValue and unformattedValue should hold the raw number or null. The old logic could assign the number set via v-model to formattedValue, that's wrong and I had to fix that to make TS happy.

Reply:
There is definitely a lack of discriminated union support for props type in Vue.